Implement internal math module

samuelselleck opened the pull request:
This implements an internal math module, to be exposed to users of pax, and removes the use of kurbo::Affine as the engines main transform type.

zackbrown commented on pax-engine/src/lib.rs:
```diff
@@ -2,6 +2,8 @@ pub extern crate pax_macro;
 pub use pax_macro::*;
 
 pub use pax_runtime::api;
+pub use pax_runtime::design_utils;
```
please remove

zackbrown commented on pax-runtime/src/engine/design_utils.rs:
please remove

zackbrown commented on pax-runtime/src/engine/mod.rs:
```diff
@@ -8,17 +8,20 @@ use std::rc::Rc;
     RuntimePropertiesStackFrame, TransformAndBounds,
 };
 
+pub mod design_utils;
```
please remove

zackbrown reviewed (changes requested):
Nice improvements!  Small changes requested re: `design_utils`, per team conversation.  After that, please go ahead and merge directly.

samuelselleck commented:
This is done now! 
- Created a new NodeInterface type that is a public wrapper over ExpandedNode
- Added methods on NodeContext to do raycasting, finding by id, etc (forwarding calls to render_context and wrapping results in NodeInterface)
- Renamed NodeContext to EngineContext to better reflect it's ability to do things like global raycasting

warfaj commented:
LGTM! Thanks for the changes.
